torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -487,6 +489,21 @@ c10::intrusive_ptr<c10d::Work> allgather_coalesced_privateuse1_(
           input_list_vec);
 }
 
+#define IMPL_ALLGATHER_INTO_TENSOR_COALESCED(DEV)                          \
+  c10::intrusive_ptr<c10d::Work> allgather_into_tensor_coalesced_##DEV##_( \
+      at::TensorList outputs,                                              \
+      at::TensorList inputs,                                               \
+      const c10::intrusive_ptr<ProcessGroup>& process_group) {             \
+    auto output_vec = outputs.vec();                                       \
+    auto input_vec = inputs.vec();                                         \
+    return process_group->getBackend(c10::DeviceType::DEV)                 \
+        ->allgather_into_tensor_coalesced(output_vec, input_vec);          \
+  }
+
+IMPL_ALLGATHER_INTO_TENSOR_COALESCED(CPU)
```

Comment:
curious shall we register impl for CPU backend? I thought it only works for CUDA/NCCL backend?

Reply:
This should not make much difference in terms of user experience.
If we don't register and user called into it, then dispatcher would complain.
If we register, then the corresponding backend will tell the user it doesn't have an implementation.
I think in the second case the error message would be slightly clearer.